pkg/config/messages.go

```diff
@@ -121,6 +127,14 @@ func (c *MessagesConfig) ValidateAndAdjust() error {
 		return cerrors.ErrInvalidServerOption.GenWithStackByArgs("server-worker-pool-size is larger than 32")
 	}
 
+	if c.MaxRecvMsgSize == 0 {
+		c.MaxRecvMsgSize = defaultMaxRecvMsgSize
+	}
+	if c.MaxRecvMsgSize <= 0 {
```

Comment:
```suggestion
	if c.MaxRecvMsgSize < 0 {
```

Reply:
It looks this 2 case can be switched, check `<0` and return the error first, then check if it's `0`.